packages/core/lib/segments/attributes/subsegment.js

```diff
@@ -379,12 +371,16 @@ Subsegment.prototype.toString = function toString() {
 };
 
 Subsegment.prototype.toJSON = function toJSON() {
-  var ignore = ['segment', 'parent', 'counter'];
+  var thisCopy = Object.assign({}, this);
+
+  if ('segment' in thisCopy) delete thisCopy.segment;
+  if ('parent' in thisCopy) delete thisCopy.parent;
+  if ('counter' in thisCopy) delete thisCopy.counter;
 
-  if (isEmpty(this.subsegments))
-    ignore.push('subsegments');
+  if (Array.isArray(this.subsegments) && !this.subsegments.length && 'subsegments' in thisCopy)
+    delete thisCopy.subsegments;
```

Comment:
Nit: Since the body of the if statement is on a separate line, can you wrap this in brackets? That'll make it more clear what our intent is.